Lib/multiprocessing/shared_memory.py

```diff
@@ -89,15 +87,15 @@ class WindowsNamedSharedMemory:
 
     def __init__(self, name, flags=None, mode=384, size=0, read_only=False):
         if name is None:
-            name = f'wnsm_{os.getpid()}_{random.randrange(100000)}'
+            name = _make_filename()
 
         if size == 0:
             # Attempt to dynamically determine the existing named shared
             # memory block's size which is likely a multiple of mmap.PAGESIZE.
             try:
                 h_map = kernel32.OpenFileMappingW(FILE_MAP_READ, False, name)
-            except OSError as ose:
-                raise ExistentialError(*ose.args)
+            except OSError:
+                raise FileNotFoundError(name)
```

Comment:
This is too broad. Suggestion: look for specific errno (not tested):


```python
try:
    h_map = kernel32.OpenFileMappingW(FILE_MAP_READ, False, name)
except OSError as err:
    if err.errno == errno.ENOENT:
        raise FileExistsError(errno.ENOENT, os.strerror(errno.ENOENT), name)
    raise
```

Reply:
The try/except shouldn't be necessary anymore, depending on the code in _winapi.

Reply:
Good point -- it was nice to eliminate the try/except in commit 68785333a25d902a42252ace3a8268357efd6094.